Posts/Article.0001.ChangeDriveLetter.md

```diff
@@ -0,0 +1,231 @@
+---
+post_title: Changing Drive Letters and Labels via PowerShell
+username: DoctorDNS
+Catagories: PowerShell, WMI
+Summary: Shows how to change a drive's letter and label using PowerShell and WMI
+---
+
+# Change Drive Letters and Labels via PowerShell
+
+
+**Q:** I want to change the drive letter and the drive label for a new USB drive. 
+Is there a way with PowerShell>
+
+**A:**  Of course. You can use WMI and the CIM cmdlets.
+
+PowerShell does not have a cmdlet to change the drive letter or the caption directly. 
+But the good news is that you can use WMI and the CIM cmdlets to change both the drive letter and drive label.
+
+## WMI Classes, Class properties and Class Methods
+
+WMI holds a huge amount of information about a Windows host in the form of WMI classes.
+Every IT professional should know about WMI.
+
+WMI holds a hierarchical database of classes and class occurrences. 
+These classes describe the hardware and software in your computer.
+This database is organized in to namespaces each which contains classes and, optionally, additional namespaces.
+You can use the CIM cmdlets to both retrieve and update this information. 
+
+For example, you can discover the drive letter and drive label for a drive from the ``Win32_Volume`` class.
+This class in in the root\CimV2 namespace.
+
+Many WMI classes also contain methods that you can use to act on the WMI object.
+You can use the ``Format()`` method of the **Win32_Volume** class to format a Windows volume.
+
+To obtain the values of the properties of a WMI class, or to invoke a class method, you can use the WMI cmdlets, which shipped with Windows PowerShell V1.
+However, these cmdlets no longer ship with PowerShell 7.
+Of course, a determined IT Pro could find a way around that - but you don't have to!
+
+With PowerShell 7, you use the CIM cmdlets to access this information.
+The CIM cmdlets first shipped with Windows PowerShell V3 and represented a major overhaul in how IT Pros access WMI.
+The newer cmdlets do the same job as the WMI cmdlets but have different cmdlets, and different ways of working as you can see in this article.
+
+
+## Discovering WMI Class Properties
+
+You use the cmdlet ``Get-CimClass`` to discover the names (and type) of the properties of any given class.
+You can discover the properties of the **Win32_Volume** class like this:
+
+```powershell
+
+Get-CimClass -ClassName Win32_Volume |
+  Select-Object -Expand CimClassProperties | 
+    Sort-Object -Property Name |
+      Format-Table Name, CimType, Qualifiers
+```
+
+The output from this commands looks like this:
+
+```powershell-console
```

Comment:
I am not sure if the new blog supports this yet. I want to use it but there may be some configuration we need to do.

Reply:
I thought that that was what the Readme.md said here?

Reply:
Yeah, this is still a work in progress. We can leave it for now.